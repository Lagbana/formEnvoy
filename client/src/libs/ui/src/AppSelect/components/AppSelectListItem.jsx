/* Entry on settings screen
=============================== */

import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Typography, Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppSelectListItemProps } from '../AppSelect.interface';

export function AppSelectListItem<T = any>({
  item,
  selected = false
}: AppSelectListItemProps<T>) {

  const styles = StyleSheet.create({
    container: {
      borderTopWidth: 1,
      borderColor: Colors.gray + '22',
      backgroundColor: Colors.grayLight,
    },
    label: {
      ...Typography.bodySmall,
      color: Colors.text,
      paddingTop: 12,
      paddingBottom: 12,
      paddingHorizontal: 20,
    },
    selectedLabel: {
      ...Typography.labelSmall,
      color: Colors.purple,
      paddingTop: 16,
      paddingBottom: 12,
      paddingHorizontal: 20,
    },
  });

  return (
    <View style={styles.container}>
      <Text style={[selected ? styles.selectedLabel : styles.label]}>
        {item.Name}
      </Text>
    </View>
  );
};
