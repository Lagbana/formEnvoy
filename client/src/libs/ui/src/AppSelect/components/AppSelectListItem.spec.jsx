import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppSelectListItem } from './AppSelectListItem'

configure({ adapter: new Adapter() })

test('<AppSelectListItem /> renders correctly', () => {
  const component = shallow(
    <AppSelectListItem
      selected={true}
			item={{ Id: 100, Name: 'selected item', Value: 100 }}
			key={'Test Key'}
    />
  )
  expect(toJson(component)).toMatchSnapshot()
})
