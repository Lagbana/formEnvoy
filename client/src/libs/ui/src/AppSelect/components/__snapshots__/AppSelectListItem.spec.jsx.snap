// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSelectListItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8FA",
      "borderColor": "#80889822",
      "borderTopWidth": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#4749A0",
          "fontFamily": "ObjektivMk1-Bold",
          "fontSize": 16,
          "lineHeight": 17.6,
          "paddingBottom": 12,
          "paddingHorizontal": 20,
          "paddingTop": 16,
        },
      ]
    }
  >
    selected item
  </Text>
</View>
`;
